docs/docs/development/new-behavior.md

```diff
@@ -0,0 +1,383 @@
+- `.locality`: Defined in `<drivers/behavior.h>`. Describes how the behavior affects parts of a _split_ keyboard.
+  - `BEHAVIOR_LOCALITY_CENTRAL`: Behavior only affects the central half
+  - `BEHAVIOR_LOCALITY_EVENT_SOURCE`: Behavior affects only the central _or_ peripheral half, depending on which side invoked the behavior binding
+  - `BEHAVIOR_LOCALITY_GLOBAL`: Behavior affects the entire keyboard
+    :::note
+    For unibody keyboards, all locality values perform the same as `BEHAVIOR_LOCALITY_GLOBAL`.
+    :::
+
+##### Data Pointers (Optional)
+
+The data `struct` stores additional data required for **each new instance** of the behavior. Regardless of the instance number, `n`, `behavior_<behavior_name>_data_##n` is typically initialized as an empty `struct`. The data respective to each instance of the behavior can be accessed in functions like [`on_<behavior_name>_binding_pressed(struct zmk_behavior_binding *binding, struct zmk_behavior_binding_event event)`](#dependencies) by extracting the behavior device from the keybind like so:
+
+```
+const struct device *dev = device_get_binding(binding->behavior_dev);
+struct behavior_<behavior_name>_data *data = dev->data;
+```
+
+The variables stored inside the data `struct`, `data`, can be then modified as necessary.
+
+The fourth cell of `DEVICE_DT_INST_DEFINE` can be set to `NULL` instead if instance-specific data is not required.
+
+##### Configuration Pointers (Optional)
+
+The configuration `struct` stores the properties declared from the behavior's `.yaml` for **each new instance** of the behavior. As seen in the `#define KP_INST(n)` of the hold-tap example, the configuration `struct`, `behavior_<behavior_name>_config_##n`, for each instance number, `n`, can be initialized using the [Zephyr Devicetree Instance-based APIs](https://docs.zephyrproject.org/latest/build/dts/api/api.html#instance-based-apis), which extract the values from the `properties` of each instance of the [devicetree binding](#creating-the-devicetree-binding-yaml) from a user's keymap or [predefined use-case `.dtsi` files](#defining-common-use-cases-for-the-behavior-dtsi-optional) stored in `app/dts/behaviors/`. We illustrate this further by comparing the [`#define KP_INST(n)` from the hold-tap driver](#device_dt_inst_define) and the [`properties` of the hold-tap devicetree binding.](#creating-the-devicetree-binding-yaml)
+
+The fifth cell of `DEVICE_DT_INST_DEFINE` can be set to `NULL` instead if instance-specific configurations are not required.
+
+:::caution
+Remember that `.c` files should be formatted according to `clang-format` to ensure that checks run smoothly once the Pull Request is submitted.
+:::
+
+### Updating `app/CmakeLists.txt` to include the new driver
+
+Behavior drivers are stored after the line `if ((NOT CONFIG_ZMK_SPLIT) OR CONFIG_ZMK_SPLIT_BLE_ROLE_CENTRAL)` in the form, `app PRIVATE src/behaviors/<behavior_name>.c`, as shown below.
```

Comment:
I think this depends on the locality right? This seems to apply to central locality behaviors.